Support net461

With the latest release, the support for cross-compilation of this nuget as a dependency went out the window, unfortunately. This is what happens:

```

/usr/local/share/dotnet/sdk/2.1.105/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Sdk.DefaultItems.targets(198,5): warning : A PackageReference for 'Microsoft.NETCore.App' was included in your project. This package is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see https://aka.ms/sdkimplicitrefs [/Users/h/dev/c/logary/src/Logary.CSharp.Tests/Logary.CSharp.Tests.csproj]
/Users/h/dev/c/logary/src/legacy/Logary.Services.Rutta/Logary.Services.Rutta.fsproj : error NU1202: Package dotnet-proj-info 0.9.0 is not compatible with net461 (.NETFramework,Version=v4.6.1). Package dotnet-proj-info 0.9.0 supports: [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Services.Rutta/Logary.Services.Rutta.fsproj : error NU1202:   - netcoreapp1.0 (.NETCoreApp,Version=v1.0) [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Services.Rutta/Logary.Services.Rutta.fsproj : error NU1202:   - netcoreapp2.0 (.NETCoreApp,Version=v2.0) [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Targets.Logstash.Tests/Logary.Targets.Logstash.Tests.fsproj : error NU1202: Package dotnet-proj-info 0.9.0 is not compatible with net461 (.NETFramework,Version=v4.6.1). Package dotnet-proj-info 0.9.0 supports: [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Targets.Logstash.Tests/Logary.Targets.Logstash.Tests.fsproj : error NU1202:   - netcoreapp1.0 (.NETCoreApp,Version=v1.0) [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Targets.Logstash.Tests/Logary.Targets.Logstash.Tests.fsproj : error NU1202:   - netcoreapp2.0 (.NETCoreApp,Version=v2.0) [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Targets.Logstash/Logary.Targets.Logstash.fsproj : error NU1202: Package dotnet-proj-info 0.9.0 is not compatible with net461 (.NETFramework,Version=v4.6.1). Package dotnet-proj-info 0.9.0 supports: [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Targets.Logstash/Logary.Targets.Logstash.fsproj : error NU1202:   - netcoreapp1.0 (.NETCoreApp,Version=v1.0) [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Targets.Logstash/Logary.Targets.Logstash.fsproj : error NU1202:   - netcoreapp2.0 (.NETCoreApp,Version=v2.0) [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Targets.Shipper/Logary.Targets.Shipper.fsproj : error NU1202: Package dotnet-proj-info 0.9.0 is not compatible with net461 (.NETFramework,Version=v4.6.1). Package dotnet-proj-info 0.9.0 supports: [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Targets.Shipper/Logary.Targets.Shipper.fsproj : error NU1202:   - netcoreapp1.0 (.NETCoreApp,Version=v1.0) [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Targets.Shipper/Logary.Targets.Shipper.fsproj : error NU1202:   - netcoreapp2.0 (.NETCoreApp,Version=v2.0) [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Services.Rutta.Tests/Logary.Services.Rutta.Tests.fsproj : error NU1202: Package dotnet-proj-info 0.9.0 is not compatible with net461 (.NETFramework,Version=v4.6.1). Package dotnet-proj-info 0.9.0 supports: [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Services.Rutta.Tests/Logary.Services.Rutta.Tests.fsproj : error NU1202:   - netcoreapp1.0 (.NETCoreApp,Version=v1.0) [/Users/h/dev/logibit/logary/src/Logary.sln]
/Users/h/dev/c/logary/src/legacy/Logary.Services.Rutta.Tests/Logary.Services.Rutta.Tests.fsproj : error NU1202:   - netcoreapp2.0 (.NETCoreApp,Version=v2.0) [/Users/h/dev/logibit/logary/src/Logary.sln]


~/dev/logibit/logary(master) » mono .paket/paket.exe why dotnet-proj-info                                                                                                                                                                            h@catamorphism
Paket version 5.166.0
NuGet dotnet-proj-info - 0.9 is a transitive dependency.
It is part of following dependency chains:

-> fszmq - 12.3.1
  -> dotnet-proj-info - 0.9

Performance:
 - Runtime: 889 milliseconds
```

--- a/.paket/Paket.Restore.targets
+++ b/.paket/Paket.Restore.targets
@@ -60,6 +60,9 @@
       <Output TaskParameter="ConsoleOutput" PropertyName="PaketRestoreLockFileHash" />
     </Exec>
 
+    <!-- Debug whats going on -->
+    <Message Importance="low" Text="calling paket restore with targetframework=$(TargetFramework) targetframeworks=$(TargetFrameworks)" />
+
     <PropertyGroup Condition="Exists('$(PaketRestoreCacheFile)') ">
       <!-- if no hash has been done yet fall back to just reading in the files and comparing them -->
       <PaketRestoreCachedHash Condition=" '$(PaketRestoreCachedHash)' == '' ">$([System.IO.File]::ReadAllText('$(PaketRestoreCacheFile)'))</PaketRestoreCachedHash>
@@ -77,9 +80,11 @@
     <!-- Step 2 Detect project specific changes -->
     <ItemGroup>
       <MyTargetFrameworks Condition="'$(TargetFramework)' != '' " Include="$(TargetFramework)"></MyTargetFrameworks>
-      <MyTargetFrameworks Condition="'@(TargetFrameworks)' != '' " Include="$(TargetFrameworks)"></MyTargetFrameworks>
+      <!-- Don't include all frameworks when msbuild explicitly asks for a single one -->
+      <MyTargetFrameworks Condition="'$(TargetFrameworks)' != '' AND '$(TargetFramework)' == '' " Include="$(TargetFrameworks)"></MyTargetFrameworks>
       <PaketResolvedFilePaths Include="@(MyTargetFrameworks -> '$(MSBuildProjectDirectory)\obj\$(MSBuildProjectFile).%(Identity).paket.resolved')"></PaketResolvedFilePaths>
     </ItemGroup>
+    <Message Importance="low" Text="MyTargetFrameworks=@(MyTargetFrameworks) PaketResolvedFilePaths=@(PaketResolvedFilePaths)" />
     <PropertyGroup>
       <PaketReferencesCachedFilePath>$(MSBuildProjectDirectory)\obj\$(MSBuildProjectFile).paket.references.cached</PaketReferencesCachedFilePath>
       <!-- MyProject.fsproj.paket.references has the highest precedence -->
@@ -89,7 +94,6 @@
       <!-- paket.references -->
       <PaketOriginalReferencesFilePath Condition=" !Exists('$(PaketOriginalReferencesFilePath)')">$(MSBuildProjectDirectory)\paket.references</PaketOriginalReferencesFilePath>
       
-      <PaketResolvedFilePath>$(MSBuildProjectDirectory)\obj\$(MSBuildProjectFile).$(TargetFramework).paket.resolved</PaketResolvedFilePath>
       <DoAllResolvedFilesExist>false</DoAllResolvedFilesExist>
       <DoAllResolvedFilesExist Condition="Exists(%(PaketResolvedFilePaths.Identity))">true</DoAllResolvedFilesExist>
       <PaketRestoreRequired>true</PaketRestoreRequired>
@@ -112,7 +116,7 @@
     <!-- Step 2 b detect relevant changes in project file (new targetframework) -->
     <PropertyGroup Condition=" '$(DoAllResolvedFilesExist)' != 'true' ">
       <PaketRestoreRequired>true</PaketRestoreRequired>
-      <PaketRestoreRequiredReason>target-framework '$(TargetFramework)' or '$(TargetFrameworks)'</PaketRestoreRequiredReason>
+      <PaketRestoreRequiredReason>target-framework '$(TargetFramework)' or '$(TargetFrameworks)' files @(PaketResolvedFilePaths)</PaketRestoreRequiredReason>
     </PropertyGroup>
 
     <!-- Step 3 Restore project specific stuff if required -->
@@ -140,8 +144,9 @@
       </PaketReferencesFileLinesInfo>
       <PackageReference Include="%(PaketReferencesFileLinesInfo.PackageName)">
         <Version>%(PaketReferencesFileLinesInfo.PackageVersion)</Version>
-        <PrivateAssets Condition="%(PaketReferencesFileLinesInfo.AllPrivateAssets) == 'true'">All</PrivateAssets>
+        <PrivateAssets Condition=" ('%(PaketReferencesFileLinesInfo.AllPrivateAssets)' == 'true') Or ('$(PackAsTool)' == 'true') ">All</PrivateAssets>
         <ExcludeAssets Condition="%(PaketReferencesFileLinesInfo.AllPrivateAssets) == 'exclude'">runtime</ExcludeAssets>
+        <Publish Condition=" '$(PackAsTool)' == 'true' ">true</Publish>
       </PackageReference>
     </ItemGroup>
 

--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -1,3 +1,6 @@
+### New in 12.3.2 (2018/05/23)
+* FIX for Issue #161 - fixed regression in .NET 4.6.1 support
+
 ### New in 12.3.1 (2018/05/18)
 * FIX (untracked) NuGet package now includes XMLDocs (`fszmq.xml`), which were accidentally omitted from last release
 

--- a/build.fsx
+++ b/build.fsx
@@ -114,9 +114,6 @@ Target "AssemblyInfo" (fun _ ->
 Target "CopyBinaries" (fun _ ->
   // managed libraries
   !! "src/fszmq/bin/Release/netstandard2.0/fszmq.*" |> CopyTo "bin"
-  // support tooling
-  !! "tests/*perf*/bin/Release/net461/*.exe" |> CopyTo "bin/net461"
-  !! "tests/*perf*/bin/Release/netcoreapp2.0/*.exe" |> CopyTo "bin/netcoreapp2.0"
   // native dependencies
   !! "lib/zeromq/OSX/**/*.*" |> CopyTo "bin/OSX"
   !! "lib/zeromq/LIN/**/*.*" |> CopyTo "bin/LIN"
@@ -146,8 +143,7 @@ Target "RunTests" (fun _ ->
     { p with 
         Project = proj
         Configuration = "Release"
-        AdditionalArgs = [ "--no-build" ] }))
-)
+        AdditionalArgs = [ "--no-build" ] })))
 
 // --------------------------------------------------------------------------------------
 // Build a deployment artifacts

--- a/docs/tools/generate.fsx
+++ b/docs/tools/generate.fsx
@@ -44,7 +44,7 @@ let netStandardRef =
   let result = ExecProcessAndReturnMessages 
                   (fun p -> p.FileName  <- "dotnet"
                             p.Arguments <- "proj-info --fsc-args"
-                            p.WorkingDirectory <- "../../src/fszmq")
+                            p.WorkingDirectory <- ".")
                   (TimeSpan.FromSeconds 5.0)
   if not result.OK
     then  ""

--- a/docs/tools/paket.references
+++ b/docs/tools/paket.references
@@ -1,0 +1,2 @@
+FSharp.Core
+dotnet-proj-info

--- a/docs/tools/tools.fsproj
+++ b/docs/tools/tools.fsproj
@@ -1,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <AssemblyName>tools</AssemblyName>
+  </PropertyGroup>
+  <Import Project="..\..\.paket\Paket.Restore.targets" />
+</Project>

--- a/src/fszmq/AssemblyInfo.fs
+++ b/src/fszmq/AssemblyInfo.fs
@@ -11,13 +11,13 @@ open System.Reflection
 [<assembly: AssemblyTitleAttribute("fszmq")>]
 [<assembly: AssemblyProductAttribute("fszmq")>]
 [<assembly: AssemblyDescriptionAttribute("An MPLv2-licensed F# binding for the ZeroMQ distributed computing library.")>]
-[<assembly: AssemblyVersionAttribute("12.3.1")>]
-[<assembly: AssemblyFileVersionAttribute("12.3.1")>]
+[<assembly: AssemblyVersionAttribute("12.3.2")>]
+[<assembly: AssemblyFileVersionAttribute("12.3.2")>]
 do ()
 
 module internal AssemblyVersionInformation =
     let [<Literal>] AssemblyTitle = "fszmq"
     let [<Literal>] AssemblyProduct = "fszmq"
     let [<Literal>] AssemblyDescription = "An MPLv2-licensed F# binding for the ZeroMQ distributed computing library."
-    let [<Literal>] AssemblyVersion = "12.3.1"
-    let [<Literal>] AssemblyFileVersion = "12.3.1"
+    let [<Literal>] AssemblyVersion = "12.3.2"
+    let [<Literal>] AssemblyFileVersion = "12.3.2"

--- a/src/fszmq/paket.references
+++ b/src/fszmq/paket.references
@@ -1,2 +1,1 @@
 FSharp.Core
-dotnet-proj-info

